### PR TITLE
ci: make absolute prestate for asterisc

### PR DESCRIPTION
**Pull request by pcw109550**
**Description**

Adds CI to calculate asterisc's absolute prestate, to prepare op-e2e tests for asterisc. To do this, 
1. Build op-program and compile op-program to riscv
2. Ask asterisc binary to create absolute prestate from op-program binary.

Example pipeline at [here](https://app.circleci.com/pipelines/github/ethereum-optimism/asterisc/56/workflows/103dea74-37ca-4f84-b6e8-c4c29b5ba953?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-checks-link&utm_content=summary):

image

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+      - run:
+          name: Fetch submodules for asterisc
+          command: git submodule update --init --recursive
+      - run:
+          name: Check L1 geth version
+          command: ./ops/scripts/geth-version-checker.sh || (echo "geth version is wrong, update ci-builder"; false)
+          working_directory: rvsol/lib/optimism
+      - run:
+          name: git submodules for monorepo
+          command: make submodules
```
Comment (ImTei) [inline .circleci/config.yml:36]: Do we need this even if we run `git submodule update` with `--recursive` flag in the prior step?

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+          working_directory: rvsol/lib/optimism
+      - restore_cache:
+          name: Restore PNPM Package Cache
+          keys:
+            - pnpm-packages-v2-{{ checksum "rvsol/lib/optimism/pnpm-lock.yaml" }}
+      - restore_cache:
+          name: Restore Go modules cache for monorepo
+          # this go mod cache wil be populated from go-mod-download-monorepo step
```
Comment (ImTei) [inline .circleci/config.yml:44]:
```suggestion
          # this go mod cache will be populated from go-mod-download-monorepo step
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+          key: pnpm-packages-v2-{{ checksum "rvsol/lib/optimism/pnpm-lock.yaml" }}
+          paths:
+            - "rvsol/lib/optimism/node_modules"
+      - run:
+          name: Install dependencies
+          command: pnpm install:ci:offline
+          working_directory: rvsol/lib/optimism
+      - run:
+          name: print forge version
+          command: forge --version
```
Comment (ImTei) [inline .circleci/config.yml:67]: Is this step just for logging?

Comment (pcw109550) [reply .circleci/config.yml:67]: Yes. I think It is worth to check foundry version, and monorepo does it also.

File: .circleci/config.yml (hunk 1 above)
Comment (pcw109550) [reply .circleci/config.yml]: No we do not. Removed

File: .circleci/config.yml (hunk 2 above)
Comment (pcw109550) [reply .circleci/config.yml]: Fixed.

**Review by ImTei — APPROVED** [review]

**Pull request merged**